### DISCUSSION TITLE
Exclude duplicate mods during import

### DISCUSSION THREAD

**Discussion by Nilruin** [xivdev/Penumbra#354] [Ideas]
It'd be nice to include a way to exclude mod files that are already installed and enabled during the import process. I just went through the process of removing 200+ mods because apparently a bunch got missed because of date sorting issues, so I decided to just install all the ones in my directory at once expecting Penumbra to throw up a bunch of duplicate errors or something. Nope, it installed them all again anyway.

There's also no way to shift-click multiple mods at once, which would've made the process easier. 

If this is already a feature, I'm not seeing it in the general settings menu or advanced settings menu. All I see is an option to "auto deduplicate on import", which doesn't do what I'm suggesting.

**Comment by Ottermandias** [comment]
You can Control-click multiple mods.
And no.

**Comment by Nilruin** [comment]
Not very user-intuitive considering that Windows uses Shift-Click for multiple selection entries. Good to know for future reference, however.

Is there a technical reason why Penumbra can't detect duplicate mods that are already installed? Most mod managers have that kind of functionality in place.

**Reply by Ottermandias** [reply]
Windows actually uses Control-Click for multiple selection entries. Shift is for selecting from-to, Ctrl is for adding to selection. The latter is what Penumbra allows, the former is technically not feasible.
Mods can have arbitrary names. Duplicate mods exist. A lot. Overwriting shared files in an existing mod can break both the installed mod and the new mod. Deleting the existing mod and installing the new one deletes stuff irreplacably. Both proved to be more problematic than redundant installations for users.
Additionally, people actively want to have the same mod installed multiple times too.
